src/distPlus/distribution/Continuous.re

```diff
@@ -3,30 +3,45 @@ open Distributions;
 type t = DistTypes.continuousShape;
 let getShape = (t: t) => t.xyShape;
 let interpolation = (t: t) => t.interpolation;
-let make = (interpolation, xyShape, knownIntegralSum): t => {
+let make = (interpolation, xyShape, integralSumCache, integralCache): t => {
   xyShape,
   interpolation,
-  knownIntegralSum,
+  integralSumCache,
+  integralCache,
 };
-let shapeMap = (fn, {xyShape, interpolation, knownIntegralSum}: t): t => {
+let shapeMap = (fn, {xyShape, interpolation, integralSumCache, integralCache}: t): t => {
   xyShape: fn(xyShape),
   interpolation,
-  knownIntegralSum,
+  integralSumCache,
+  integralCache,
 };
 let lastY = (t: t) => t |> getShape |> XYShape.T.lastY;
 let oShapeMap =
-    (fn, {xyShape, interpolation, knownIntegralSum}: t)
+    (fn, {xyShape, interpolation, integralSumCache, integralCache}: t)
     : option(DistTypes.continuousShape) =>
-  fn(xyShape) |> E.O.fmap(make(interpolation, _, knownIntegralSum));
+  fn(xyShape) |> E.O.fmap(make(interpolation, _, integralSumCache, integralCache));
 
+let emptyIntegral: DistTypes.continuousShape = {
+  xyShape: {xs: [|neg_infinity|], ys: [|0.0|]},
+  interpolation: `Linear,
+  integralSumCache: Some(0.0),
+  integralCache: None,
+};
 let empty: DistTypes.continuousShape = {
   xyShape: XYShape.T.empty,
   interpolation: `Linear,
-  knownIntegralSum: Some(0.0),
+  integralSumCache: Some(0.0),
+  integralCache: Some(emptyIntegral),
 };
+
+let stepwiseToLinear = (t: t): t =>
+  make(`Linear, XYShape.Range.stepwiseToLinear(t.xyShape), t.integralSumCache, t.integralCache);
+
 let combinePointwise =
     (
-      ~knownIntegralSumsFn,
+      ~integralSumCachesFn=(_, _) => None,
+      ~integralCachesFn: (t, t) => option(t) =(_, _) => None,
```

Comment:
I'm curious why this is ``Caches`` and not ``Cache``?

Reply:
It's a function that combines two caches, as opposed to a function that transforms a single cache (like in `mapY`) :man_shrugging: I'm not attached to the name though if you prefer the singular